src/utils/bump.js

```diff
@@ -0,0 +1,186 @@
+'use strict'
+
+const semver = require('semver')
+const { logError } = require('../log')
+const {
+  parser,
+  toConventionalChangelogFormat,
+} = require('@conventional-commits/parser')
+
+async function getAutoBumpedVersion({ github, context }) {
+  const { owner, repo } = context.repo
+
+  const {
+    latestReleaseCommitSha,
+    latestReleaseTagName,
+    latestReleaseCommitDate,
+  } = await getLatestRelease({ github, owner, repo })
+
+  if (
+    !latestReleaseCommitSha ||
+    !latestReleaseTagName ||
+    !latestReleaseCommitDate
+  ) {
+    throw new Error(`Couldn't find latest release`)
+  }
+
+  const allCommits = await getCommitMessagesSinceLatestRelease({
+    github,
+    owner,
+    repo,
+    commitDate: latestReleaseCommitDate,
```

Comment:
If we use the commit sha then using the graphql API there is no easy way to get the list of commits after a particular sha. I will explore a bit more to see how we can avoid using `since` and do it using the sha.

> commits with an older date could land in the repository after commits with a newer date
I _guess_ using `since` should take care of this automatically but I will try this out to see how the code behaves in this scenario